src/twisted/internet/test/test_iocp.py

```diff
@@ -94,21 +94,22 @@ def _acceptAddressTest(self, family, localhost):
                     SOL_SOCKET, SO_UPDATE_ACCEPT_CONTEXT, pack("P", port.fileno())
                 )
                 # Reset any previous error and stop retrying.
-                socketError = None
+                lastError = None
                 break
             except OSError as socketError:
+                lastError = socketError
                 # getattr is used below to make mypy happy.
-                if socketError.errno == getattr(errno, "WSAENOTCONN"):
+                if lastError.errno == getattr(errno, "WSAENOTCONN"):
                     # Without a sleep here even retrying 20 times will fail.
                     # This should allow other threads to execute.
                     time.sleep(0.2)
                 else:
                     # This is not the expected error so re-raise the error without retrying.
                     raise
 
-        if socketError:
+        if lastError:
```

Comment:
I think it's neater to count backwards, than assign and re-assign the error:

```python
for attempts_remaining in range(4, -1, -1):
    try:
        server.setsockopt(SOL_SOCKET, SO_UPDATE_ACCEPT_CONTEXT, pack("P", port.fileno()))
        break
    except OSError as e:
        if attempts_remaining == 0 or e.errno != getattr(errno, "WSAENOTCONN"):
            raise
        time.sleep(0.2)
```

Reply:
yep. Will do it. thanks :)